### DISCUSSION TITLE
Contribution Guidance

### DISCUSSION THREAD

**Discussion by chinmoy12c** [ACINQ/lightning-kmp#324] [Q&A] [answered]
Hello everyone, I've recently started learning about bitcoin and have stumbled upon the lightning-kmp repo. I'm planning to contribute to this project because I find it a really promising and necessary endeavor. I have a good amount of experience in native android development but not much in kmp. I was wondering if there are any resources that could help me get up to speed with the codebase (for example, some documentation on how parts of the codebase are designed). It would be of great help if someone could direct me to some good docs.

Thank you :)

**Comment by t-bast** [comment, marked as answer]
Welcome! We don't really have any design doc, this codebase has a lot of moving parts as we're also learning about kmp.
Before you contribute, you should probably have a good understanding of the lightning protocol.
Good starting points for this are:

- https://lightning.network/lightning-network-paper.pdf
- https://github.com/ElementsProject/lightning/raw/master/doc/deployable-lightning.pdf
- https://github.com/lightning/bolts
- https://github.com/t-bast/lightning-docs/

There is a lot to learn in those documents, so don't expect to digest it all at once.
You can just skim through them at first, and then go back for a more in-depth look once you need it.

Then you'll need to figure out what exactly you want to contribute to: is there a specific feature you'd like to add?

**Reply by chinmoy12c** [reply]
Thanks, I'll take a look at these :). Actually, I'm applying for summer of bitcoin this year so I was planning to make myself comfortable with the codebase early and start with some small PRs which get some of the work done. This would also help ease the workload later in the main development phase. So, if there are some specific things I need to go through regarding those ideas(for example, multiple peers support), please do list them :).

**Reply by t-bast** [reply]
Great! I don't think we have small tasks to do in the codebase to ramp up at the moment unfortunately, unless there are small spec features that we haven't implemented yet. But we do have a great long term feature to add multiple peers support as I highlighted in the summer of bitcoin description, that's a good project to tackle. It shouldn't require too much prior knowledge about lightning since we already support having a peer, it's mostly some refactoring and architecture improvements to go from a single peer to many. You could start prototyping this, and then extract it into small PRs that we could integrate one by one.
